composer/workflows/create_asynchronous_dataform_workflow_invocation.py

```diff
@@ -0,0 +1,87 @@
+# limitations under the License.
+
+# [START composer_dataform_async_workflow_invocation]
+"""
+Example Airflow DAG that creates a Dataform compilation result, starts an asynchronous Dataform workflow invocation, and polls the status of the workflow until it enters a desired state by using DataformWorkflowInvocationStateSensor.
+This Airflow DAG uses Google Dataform Airflow operators. For more information about Google Dataform Airflow operators, see https://airflow.apache.org/docs/apache-airflow-providers-google/stable/operators/cloud/dataform.html?highlight=dataform#google-dataform-operators
+"""
+
+import datetime
+
+from airflow import models
+from airflow.providers.google.cloud.operators.dataform import (
+    DataformCreateCompilationResultOperator,
+    DataformCreateWorkflowInvocationOperator,
+)
+from airflow.providers.google.cloud.sensors.dataform import (
+    DataformWorkflowInvocationStateSensor,
+)
+from google.cloud.dataform_v1beta1 import WorkflowInvocation
+
+
+DAG_ID = "dataform"
+PROJECT_ID = "my_project_ID"  # Replace with your Dataform Google Cloud Project ID
+REPOSITORY_ID = "my_repository_ID"  # Replace with the name of your Dataform repository
+REGION = (
+    "us-central1"  # Replace with the region in which the Dataform repository is located
+)
+GIT_COMMITISH = (
+    "main"  # Replace with the Git branch or a Git SHA in your remote Git repository
+)
+
+with models.DAG(
+    DAG_ID,
+    schedule_interval="@once",  # Override to match your needs
+    start_date=datetime.datetime(2022, 1, 1),
+    catchup=False,  # Override to match your needs
+    tags=["dataform"],
+) as dag:
+
+    create_compilation_result = DataformCreateCompilationResultOperator(
+        task_id="create_compilation_result",
+        project_id=PROJECT_ID,
+        region=REGION,
+        repository_id=REPOSITORY_ID,
+        compilation_result={
+            "git_commitish": GIT_COMMITISH,
+        },
+    )
+
+create_workflow_invocation = DataformCreateWorkflowInvocationOperator(
+    task_id="create_workflow_invocation",
+    project_id=PROJECT_ID,
+    region=REGION,
+    repository_id=REPOSITORY_ID,
+    asynchronous=True,
+    workflow_invocation={
+        "compilation_result": "{{ task_instance.xcom_pull('create_compilation_result')['name'] }}"
+    },
+)
+
+is_workflow_invocation_done = DataformWorkflowInvocationStateSensor(
+    task_id="is_workflow_invocation_done",
+    project_id=PROJECT_ID,
+    region=REGION,
+    repository_id=REPOSITORY_ID,
+    # workflow_invocation_id is last element of full resource name generated in create_workflow_invocation
+    workflow_invocation_id=(
+        "{{ task_instance.xcom_pull('create_workflow_invocation')['name'].split('/')[-1] }}"
+    ),
+    expected_statuses={WorkflowInvocation.State.SUCCEEDED},
+)
+
+# NOTE: is_workflow_invocation_done waits for workflow completion, it might take significant amount of time to finish depending on the project
```

Comment:
Is there a rough rough estimate you can give (i.e. more than an hour)?

Reply:
@mizeram yeah, this is a fair ask from Dan - it's good to give users an idea of if they should be expecting minutes/hours/days - significant means something different to everyone 😄

Reply:
We anticipate a huge range of execution times, starting from seconds (for new projects doing  simple queries on small datasets) to even days (large, complex projects running queries on big datasets). I would be reluctant to give any indication how long executions will take in such a comment, as it would end up misleading for at least some customers. Customers can always check progress of particular execution and times of past executions of given project through both UI and API. 

I would assume customers have already run the project before executing it through this code sample and they will know the approximate execution time. 

Reply:
If there is a way for the user to somewhat check the progress then that sounds good to me. I wanted the users to not have to wait endlessly with no prompts and not knowing when it would finish, which would lead to further frustration and confusion. Sounds like this is not the case, thank you for the clarification :)